### PR TITLE
SameInstance nested content bug fix

### PR DESCRIPTION
**Pull request by aurislt7**

### CODE REVIEW

File: presentation/common/src/presentation-common/content/ContentTraverser.ts [hunk 1]
```diff
@@ -299,29 +299,34 @@ function traverseContentItemPrimitiveFieldValue(visitor: IContentVisitor, fieldH
   visitor.processPrimitiveValue({ field: fieldHierarchy.field, valueType, namePrefix, rawValue, displayValue });
 }
 
-function createFieldHierarchies(fields: Field[]) {
+/**
+ * `ignoreCategories` parameter enables adding all of the `nestedFields` to parent field's `childFields`
+ *  without considering categories.
+ * @alpha
```
Comment (grigasp) [inline presentation/common/src/presentation-common/content/ContentTraverser.ts:305]: why not `@internal`?

File: presentation/common/src/presentation-common/content/ContentTraverser.ts [hunk 2]
```diff
@@ -299,29 +299,34 @@ function traverseContentItemPrimitiveFieldValue(visitor: IContentVisitor, fieldH
...
+*/
+export function createFieldHierarchies(fields: Field[], ignoreCategories?: Boolean) {
   const hierarchies = new Array<FieldHierarchy>();
-  const visitField = (category: CategoryDescription, field: Field, parentField: Field | undefined): FieldHierarchy | undefined => {
+  const visitField = (category: CategoryDescription, field: Field, parentField: Field | undefined, ignoreCategoriesFlag?: Boolean): FieldHierarchy | undefined => {
     let childFields: FieldHierarchy[] = [];
     if (field.isNestedContentField()) {
       // visit all nested fields
-      childFields = visitFields(field.nestedFields, field);
+      childFields = visitFields(field.nestedFields, field, ignoreCategories);
```
Comment (grigasp) [inline presentation/common/src/presentation-common/content/ContentTraverser.ts:313]: The flag is available in scope, right? I don't think we need to pass it into these nested functions.

File: presentation/components/src/presentation-components/table/DataProvider.ts [hunk 3]
```diff
@@ -439,6 +439,8 @@ class CellsBuilder extends PropertyRecordsBuilder {
         const expandedNestedContentField = this._sameInstanceFields[hierarchy.field.name];
         expandedNestedContentField.name = hierarchy.field.name;
         hierarchy.field = expandedNestedContentField;
+        const updatedField = createFieldHierarchies([expandedNestedContentField], true);
```
Comment (grigasp) [inline presentation/components/src/presentation-components/table/DataProvider.ts:442]: I suggest renaming the variable to `updatedFieldHierarchy`.

File: full-stack-tests/presentation/src/frontend/providers/TableDataProvider.test.ts [hunk 4]
```diff
@@ -167,7 +167,7 @@ describe("TableDataProvider", async () => {
           value: {
             valueFormat: PropertyValueFormat.Array,
             itemsTypeName: "Physical Object",
-            items: (value: PropertyRecord[]) => value.length === 2,
+            items: (value: PropertyRecord[]) => value.length === 2 && value.every((item) => Object.keys((item.value as StructValue).members).length > 0),
```
Comment (grigasp) [inline full-stack-tests/presentation/src/frontend/providers/TableDataProvider.test.ts:170]: Kind of hard to understand when the check is doing. I suggest adding a function `isStructWithMembers(record: PropertyRecord): boolean` and then using it here `value.every(isStructWithMembers)`.

**Review by grigasp — CHANGES_REQUESTED** [review]

File: presentation/common/src/presentation-common/content/ContentTraverser.ts (hunk 1 above)
Comment (aurislt7) [reply presentation/common/src/presentation-common/content/ContentTraverser.ts]: Updated

File: presentation/common/src/presentation-common/content/ContentTraverser.ts (hunk 2 above)
Comment (aurislt7) [reply presentation/common/src/presentation-common/content/ContentTraverser.ts]: Updated

File: presentation/components/src/presentation-components/table/DataProvider.ts (hunk 3 above)
Comment (aurislt7) [reply presentation/components/src/presentation-components/table/DataProvider.ts]: Done

File: full-stack-tests/presentation/src/frontend/providers/TableDataProvider.test.ts (hunk 4 above)
Comment (aurislt7) [reply full-stack-tests/presentation/src/frontend/providers/TableDataProvider.test.ts]: Done

**Review by grigasp — APPROVED** [review]

**Pull request merged**